src/Slices/ServiceInstanceDetails/UI/Components/InstanceActions/Actions/DestroyAction.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Alert, DropdownItem, Text } from "@patternfly/react-core";
+import { TrashAltIcon } from "@patternfly/react-icons";
+import { ParsedNumber } from "@/Core";
+import { useDestroyInstance } from "@/Data/Managers/V2/DELETE/DestroyInstance";
+import { DependencyContext, words } from "@/UI";
+import { ConfirmationModal } from "../../ConfirmModal";
+import { ToastAlertMessage } from "../../ToastAllert";
+
+interface Props {
+  instance_display_identity: string;
+  instance_id: string;
+  service_entity: string;
+  version: ParsedNumber;
+  onClose: () => void;
+  setInterfaceBlocked: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+/**
+ * The DestroyAction Component
+ *
+ * @props {Props} props - The props of the components
+ *  @prop {string} instance_display_identity - the display value of the instance Id
+ *  @prop {string} instance_id - the hashed id of the instance
+ *  @prop {string} service_entity - the service entity type of the instance
+ *  @prop {ParsedNumber} version - the current version of the instance
+ *  @prop {function} onClose - callback method when the modal gets closed
+ *  @prop {React.Dispatch<React.SetStateAction<boolean>>} setInterfaceBlocked - setState variable to block the interface when the modal is opened.
+ *  This is meant to avoid clickEvents triggering the onOpenChange from the dropdown to shut down the modal.
+ * @returns {React.FC<Props>} A React Component displaying the Destroy Dropdown Item
+ */
+export const DestroyAction: React.FC<Props> = ({
+  service_entity,
+  instance_display_identity,
+  instance_id,
+  version,
+  onClose,
+  setInterfaceBlocked,
+}) => {
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string>("");
+
+  const navigate = useNavigate();
+
+  const { environmentHandler, authHelper } = useContext(DependencyContext);
+
+  const environment = environmentHandler.useId();
+  const username = authHelper.getUser();
+  const message = words("instanceDetails.API.message.update")(username);
+
+  const { mutate, isError, error, isSuccess, isPending } = useDestroyInstance(
+    environment,
+    instance_id,
+    service_entity,
+    version,
+    message,
+  );
+
+  /**
+   * When the destroy action is selected, block the interface and open the modal
+   */
+  const onDestroySelect = () => {
+    setInterfaceBlocked(true);
+    setIsModalOpen(true);
+  };
+
+  /**
+   * shorthand method to handle the state updates when the modal is closed.
+   */
+  const closeModal = useCallback(() => {
+    setIsModalOpen(false);
+    setInterfaceBlocked(false);
+    onClose();
+  }, [setIsModalOpen, setInterfaceBlocked, onClose]);
+
+  /**
+   * async method sending out the request to destroy the instance
+   */
+  const onSubmitDestroy = async (): Promise<void> => {
+    mutate("");
```

Comment:
I remember you talking that you couldn't make mutate with no arguments, could you point where you found that info/example? I found a issue from 2022 where someone from RQ team said it should be possible

Reply:
It will work if empty, but the TS linter will not accept it